vaadin-card-flow-parent/vaadin-card-flow/src/test/java/com/vaadin/flow/component/card/CardTest.java

```diff
@@ -0,0 +1,330 @@
+    public void cardWithoutContent_removeNull_throwsNullPointerException() {
+        card.remove(Collections.singletonList(null));
+    }
+
+    @Test
+    public void removeAll_allChildrenRemoved() {
+        card.add(new Div(), new Div());
+        card.removeAll();
+        Assert.assertTrue(card.getChildren().findAny().isEmpty());
+    }
+
+    @Test
+    public void emptyCard_addComponentAtIndex_componentAddedAtCorrectIndex() {
+        var component = new Div();
+        card.addComponentAtIndex(0, component);
+        var firstComponent = card.getChildren().findFirst();
+        Assert.assertTrue(firstComponent.isPresent());
+        Assert.assertEquals(component, firstComponent.get());
+    }
+
+    @Test
+    public void addComponentAtNextIndex_componentAddedAtCorrectIndex() {
+        card.add(new Span());
+        var component = new Div();
+        var initialCount = (int) card.getChildren().count();
+        card.addComponentAtIndex(initialCount, component);
+        var firstComponent = card.getChildren().skip(initialCount).findFirst();
+        Assert.assertTrue(firstComponent.isPresent());
+        Assert.assertEquals(component, firstComponent.get());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addComponentAtOutOfBoundsIndex_throwsIllegalArgumentException() {
+        card.add(new Span());
+        var component = new Div();
+        card.addComponentAtIndex((int) card.getChildren().count() + 1,
+                component);
+    }
+
+    @Test
+    public void addComponentAtIndex_componentsAddedAtCorrectIndexes() {
+        card.add(new Span());
+        var component = new Div();
+        card.addComponentAtIndex(0, component);
+        var firstComponent = card.getChildren().findFirst();
+        Assert.assertTrue(firstComponent.isPresent());
+        Assert.assertEquals(component, firstComponent.get());
+    }
+
+    @Test
+    public void cardThemeVariantsEmptyByDefault() {
+        Assert.assertTrue(card.getThemeNames().isEmpty());
+    }
+
+    @Test
+    public void ariaRoleEmptyByDefault() {
+        Assert.assertTrue(card.getAriaRole().isEmpty());
+    }
+
+    @Test
+    public void setAriaRole_ariaRoleUpdated() {
+        var ariaRole = "custom-role";
+        card.setAriaRole(ariaRole);
+        Assert.assertTrue(card.getAriaRole().isPresent());
+        Assert.assertEquals(ariaRole, card.getAriaRole().get());
+    }
+
+    @Test
+    public void setAriaRoleNull_ariaRoleUpdated() {
+        card.setAriaRole("custom-role");
+        card.setAriaRole(null);
+        Assert.assertTrue(card.getAriaRole().isEmpty());
+    }
+
+    private void setSlotContent_slotContentIsSet(
+            BiConsumer<Card, Component> setter, String slotName) {
+        var slotContent = new Div();
+        setter.accept(card, slotContent);
+        Assert.assertEquals(slotName,
+                slotContent.getElement().getAttribute("slot"));
+    }
+
+    private void slotBasedFieldUpdatedCorrectly(
```

Comment:
The test should probably also check that the component is actually added as a child, and is then removed when setting `null`.

The tests for the main content slot technically also don't test whether components are ever added to the UI or not.

Reply:
Added assertions that check if the children are attached and have the card as an ancestor.